pkg/apis/extensions/v1beta1/types.go

```diff
@@ -286,6 +290,43 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Conditions represent the latest available observations of a deployment's current state.
+	Conditions []DeploymentCondition `json:"conditions,omitempty"`
+}
+
+type DeploymentConditionType string
+
+// These are valid conditions of a deployment.
+const (
+	// DeploymentProgressing means the deployment is progressing.
+	DeploymentProgressing DeploymentConditionType = "Progressing"
+	// DeploymentFailed means the deployment has failed its execution.
```

Comment:
What about in the registry? Could we compare the hashes of the current/inbound resources and clear it the condition then? I'm not sure if that's an appropriate use of the persistence layer.